### DISCUSSION TITLE
Pass dynamic arguments to a shortcut configed in contributes keybindings

### DISCUSSION THREAD

**Discussion by KrabsWong** [microsoft/vscode-discussions#396] [Extension Development QnA] [answered]
I added a keybinding config in contributes as the codes below

```json
"keybindings": [
  {
    "command": "demo.extension.saveComment",
    "key": "Cmd+1",
    "mac": "Cmd+1"
  }
]
```

It's used for saving user's comment in editor (Code review), since the save function needs some arguments to finish the save flow, is there a way to pass the dynamic argument with the config in contributes or any other way to implement it.

Currently, I can set a static value with "args" field only[https://code.visualstudio.com/docs/getstarted/keybindings#_command-arguments](https://code.visualstudio.com/docs/getstarted/keybindings#_command-arguments)

**Comment by r3m0t** [comment, marked as answer]
Make a separate command which is just for keybindings, it can use the user's activeTextEditor and selections (cursor position) to find a value for those dynamic arguments, then call the original command. If the code can't figure it out for some reason display a quick pick.

**Reply by KrabsWong** [reply]
Thanks so much for your response.

It's also my Plan B, hope there is some build-in solutions.
